packages/core/admin/admin/src/pages/Admin/index.js

```diff
@@ -4,281 +4,121 @@
-          currentLocale={locale}
-          formatMessage={formatMessage}
-          plugins={plugins}
-          shouldUpdateStrapi={shouldUpdateStrapi}
-          strapiVersion={strapiVersion}
-          updateMenu={this.state.updateMenu}
-        >
-          <Wrapper>
-            <LeftMenu
-              shouldUpdateStrapi={shouldUpdateStrapi}
-              version={strapiVersion}
-              plugins={plugins}
-              setUpdateMenu={this.setUpdateMenu}
-            />
-            <NavTopRightWrapper>
-              {/* Injection zone not ready yet */}
-              <Logout />
-            </NavTopRightWrapper>
-            <div className="adminPageRightWrapper">
-              <Header />
-              <Content>
-                <Switch>
-                  <Route path="/" render={props => this.renderRoute(props, HomePage)} exact />
-                  <Route path="/me" component={ProfilePage} />
-                  <Route path="/plugins/:pluginId" render={this.renderPluginDispatcher} />
-                  <Route path="/list-plugins" exact>
-                    <CheckPagePermissions permissions={adminPermissions.marketplace.main}>
-                      <InstalledPluginsPage />
-                    </CheckPagePermissions>
-                  </Route>
-                  <Route path="/marketplace">
-                    <CheckPagePermissions permissions={adminPermissions.marketplace.main}>
-                      <MarketplacePage />
-                    </CheckPagePermissions>
-                  </Route>
-                  <Route path="/settings/:settingId" component={SettingsPage} />
-                  <Route path="/settings" component={SettingsPage} exact />
-                  <Route key="7" path="" component={NotFoundPage} />
-                  <Route key="8" path="/404" component={NotFoundPage} />
-                </Switch>
-              </Content>
-            </div>
-
-            {process.env.STRAPI_ADMIN_SHOW_TUTORIALS === 'true' && <OnboardingVideos />}
-          </Wrapper>
-        </GlobalContextProvider>
-      </PermissionsManager>
-    );
-  }
-}
-
-Admin.defaultProps = {
-  intl: {
-    formatMessage: () => {},
-    locale: 'en',
-  },
+const HomePage = lazy(() => import(/* webpackChunkName: "Admin_homePage" */ '../HomePage'));
+const InstalledPluginsPage = lazy(() =>
+  import(/* webpackChunkName: "Admin_pluginsPage" */ '../InstalledPluginsPage')
+);
+const MarketplacePage = lazy(() =>
+  import(/* webpackChunkName: "Admin_marketplace" */ '../MarketplacePage')
+);
+const NotFoundPage = lazy(() => import('../NotFoundPage'));
+const PluginDispatcher = lazy(() =>
+  import(/* webpackChunkName: "Admin_pluginDispatcher" */ '../PluginDispatcher')
+);
+const ProfilePage = lazy(() =>
+  import(/* webpackChunkName: "Admin_profilePage" */ '../ProfilePage')
+);
+const SettingsPage = lazy(() =>
+  import(/* webpackChunkName: "Admin_settingsPage" */ '../SettingsPage')
+);
+// These are internal plugins
+const CM = lazy(() =>
```

Comment:
@alexandrebodin

Reply:
I see. I'm still not convinced by this. feels like the plugin should do it themselves

```js
import CTB from '@strapi/ctb/admin';


```

and in CTB
`index.js`
```js
export default () => {

  return {
    main: lazy(() => import('./Main'))
  }
}
```

Reply:
the CTB, CM & upload are required ones I think it is easier if we load them directly in the admin package, however it does not prevent us from keeping the actual way of loading the plugin and adding lazy.